tests/protocol/test_goal_state.py

```diff
@@ -3,17 +3,24 @@
 
 import json
 import os
+import http.client as http_client
+
 
 from azurelinuxagent.common import conf
 from azurelinuxagent.common.exception import IncompleteGoalStateError
 from azurelinuxagent.common.protocol.goal_state import GoalState, ExtensionsConfig
+from azurelinuxagent.common.protocol import hostplugin
+from azurelinuxagent.common.utils import restutil
 from tests.protocol.mocks import mock_wire_protocol
 from tests.protocol import mockwiredata
-from tests.protocol.mocks import HttpRequestPredicates
+from tests.protocol.mocks import HttpRequestPredicates, MockHttpResponse
 from tests.tools import AgentTestCase, patch
 
+_original_http_request = restutil.http_request
+
+@patch("azurelinuxagent.common.protocol.wire.conf.get_fetch_vm_settings", return_value=True)
```

Comment:
I dont see this property being used anywhere in any of the tests. If its just for the purpose of mocking, wont it be better to add it to setUp/tearDown instead?

Reply:
yes, it is just for mocking. the conf parameter is just temporary so i'm going for convenience here instead of explicit start/stop on setup/teardown